packages/tradershub/src/flows/FirstSignup/CitizenshipModal/CitizenshipModal.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from 'react';
+import { useFormikContext } from 'formik';
+import { Modal } from '@/components';
+import { isCVMEnabled } from '@/helpers';
+import { useClientCountry, useResidenceList } from '@deriv/api';
+import { LabelPairedChevronDownMdRegularIcon } from '@deriv/quill-icons';
+import { Button, Checkbox, Dropdown, Text } from '@deriv-com/ui';
+import { TFirstSignupFormValues } from '../FirstSignupWrapper/FirstSignupWrapper';
+
+type TCitizenshipModal = {
+    onClickNext: VoidFunction;
+};
+
+const CitizenshipModal = ({ onClickNext }: TCitizenshipModal) => {
+    const { data: residenceList, isLoading: residenceListLoading } = useResidenceList();
+    const { data: clientCountry, isLoading: clientCountryLoading } = useClientCountry();
+    const [isCheckBoxChecked, setIsCheckBoxChecked] = useState(false);
+    const { values, setFieldValue } = useFormikContext<TFirstSignupFormValues>();
+    const isCheckboxVisible = isCVMEnabled(values.country);
+
+    useEffect(() => {
+        if (residenceList?.length && clientCountry && values.country === '') {
+            setFieldValue('country', clientCountry);
+        }
+    }, [clientCountry, setFieldValue, residenceList, values.country]);
+
+    // Add <Loading /> here later when it's created
+    if (clientCountryLoading && residenceListLoading) return null;
+
+    return (
+        <Modal className='h-full rounded-default max-w-[328px] lg:max-w-[440px]'>
```

Comment:
Do we need this modal since youre already using react-modal, plus i dont think we need to pass width here since you already have max width in the div below

Reply:
@thisyahlen-deriv in my opinion we need this because in this component there are common styles for all modals, for example, background, corner radius, etc... But ok, I'll delete it and move the common styles inside of the div